compiler/resolution/postFold.cpp

```diff
@@ -464,6 +464,11 @@ static bool isSubTypeOrInstantiation(Type* sub, Type* super) {
   if (sub == super) {
     retval = true;
 
+  } else if (canInstantiate(sub, super)) {
+    // handles special cases like dtIntegral, which aren't covered
+    // by at->instantiatedFrom
+    retval = true;
```

Comment:
Thinking aloud, it might be useful to at some point merge canInstantiate and isSubTypeOrInstantiation into a single function